lib/spack/spack/binary_distribution.py

```diff
@@ -1345,6 +1345,10 @@ def push(specs, push_url, include_root=True, include_dependencies=True, **kwargs
         **kwargs: TODO
 
     """
+    # Be explicit about the arugment type
+    if type(include_root) != bool or type(include_dependencies) != bool:
+        raise ValueError("Expected include_root/include_dependencies to be True/False")
```

Comment:
Curious if we would have caught this with an annotated function, using `mypy`.

Reply:
It doesn't :(

Reply:
is this needed at all now that we have annotations and `mypy`?

Reply:
One other thought: I'm pretty positive this would've been caught if we checked the whole codebase with `mypy`.  Right now we only check annotated functions. I think we should try to get to checking everything to prevent subtle stuff like this -- the codebase is large enough that it can save us a lot of pain.

Reply:
This should prob use isinstance instead of bool